autorest/preparer_test.go

```diff
@@ -70,7 +70,7 @@ func ExampleCreatePreparer() {
 	} else {
 		fmt.Println(r.URL)
 	}
-	// Output: https://microsoft.com/a/b/c
```

Comment:
Is some of the confusion stemming from the fact that Request has it's own Host field that overrides the URL's Host?